src/libkomwm.py

```diff
@@ -354,11 +354,23 @@ def rule_sort_key(dict_):
                     elif st.get('-x-kot-layer') == 'bottom':
                         st['z-index'] = float(st.get('z-index', 0)) - 15001.
 
-                    if st.get('casing-width') not in (None, 0):  # and (st.get('width') or st.get('fill-color')):
+                    if st.get('casing-width') not in (None, 0) or st.get('casing-width-add') is not None:  # and (st.get('width') or st.get('fill-color')):
                         is_area_st = 'fill-color' in st
                         if has_lines and not is_area_st and st.get('casing-linecap', 'butt') == 'butt':
                             dr_line = LineRuleProto()
-                            dr_line.width = (st.get('width', 0) * WIDTH_SCALE) + (st.get('casing-width') * WIDTH_SCALE * 2)
+
+                            # 'casing-width' has precedence over 'casing-width-add'.
+                            if st.get('casing-width-add') is not None and st.get('casing-width') in (None, 0):
+                                # Look for base width in other style objects (usually ::default).
+                                base_width = 0
+                                for wst in zstyle:
+                                    if wst.get('width') not in (None, 0):
+                                        # Rail bridge styles use width from ::dash object instead of ::default.
+                                        if base_width == 0 or wst.get('object-id') != '::default':
```

Comment:
Hm, looks strange here. Why does width depend from ::dash or ::default ?

Reply:
Because railways have inner ::dash line and outer ::default line. And railway bridge evals were taking it from the first one. If there are several style object with the same param (e.g. width) then the eval() code was taking the last one - probably in order they were declared in the file (::dash lines are declared later).
I think it'll be better to hardcode it to take width from ::default always but that would mean rewriting all railway bridges casing-width-add params which were inherited from evals. And it could possibly break some other declarations that were relying on this order. So I preferred not to touch it for now.

Reply:
Hm, the logic looks very unclear here. What about taking max width always?

Reply:
Yeap its possible, an ultimate question is how we define the base/core width - is it the first/default defined line? Or the inner-most (narrowest) line? Or the outer-most (widest) one?

Reply:
ATM railways' bridge rules are based on `::dash` lines which are narrower, so we'll have to re-do them if we change to the widest line. Hence I left it as is.